test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1896,9 +1904,19 @@ protected Settings nodeSettings(int nodeOrdinal) {
 
         // Randomly set a replication strategy for the node. Replication Strategy can still be manually overridden by subclass if needed.
         if (useRandomReplicationStrategy()) {
-            ReplicationType replicationType = randomBoolean() ? ReplicationType.DOCUMENT : ReplicationType.SEGMENT;
-            logger.info("Randomly using Replication Strategy as {}.", replicationType.toString());
-            builder.put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), replicationType);
+            logger.info("Randomly using Replication Strategy as {}.", randomReplicationType.toString());
+            builder.put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), randomReplicationType);
+        }
+
+        // Randomly set storage type for the node. Storage Type can still be manually overridden by subclass if needed.
+        if (useRemoteBackedStorageRandomly()) {
```

Comment:
Do we need this extra configuration? Why not randomly use remote storage if randomReplicationType is segment?

Reply:
I don't have any strong opinion on separate configuration. Sure I think we can just use randomReplicationType and if replication type is SEGMENT, we can either pick node-node or remote store randomly

Reply:
I am thinking if there will be a use case where Tests needed only for node - node case but fail / not need for remote store case ? In that case it will be hard to test those specific Test Classes and separation of node-node and remote store will be better

Reply:
Tests that are needed for a specific replication strategy would not be using these paths anyway.  This would only be used where replication strategy is random no?

Reply:
Makes sense. Added in latest commit